lib/mixpanel-node.js

```diff
@@ -158,15 +167,111 @@ var create_client = function(token, config) {
             properties = {};
         }
 
-        if (time === void 0) {
-            throw new Error("The import method requires you to specify the time of the event");
-        } else if (Object.prototype.toString.call(time) === '[object Date]') {
-            time = Math.floor(time.getTime() / 1000);
+        properties.time = parse_time(time);
+
+        metrics.track(event, properties, callback);
+    };
+
+    /**
+        import_batch(event_list, options, callback)
+        ---
+        This function sends a list of events to mixpanel using the import
+        endpoint. The format of the event array should be:
+
+        [
+            {
+                "event": "event name",
+                "properties": {
+                    "time": new Date(), // Number or Date; required for each event
+                    "key": "val",
+                    ...
+                }
+            },
+            {
+                "event": "event name",
+                "properties": {
+                    "time": new Date()  // Number or Date; required for each event
+                }
+            },
+            ...
+        ]
+
+        See import() for further information about the import endpoint.
+
+        Options:
+            max_batch_size: the maximum number of events to be transmitted over
+                            the network simultaneously. useful for capping bandwidth
+                            usage.
+
+        N.B.: the Mixpanel API only accepts 50 events per request, so regardless
+        of max_batch_size, larger lists of events will be chunked further into
+        groups of 50.
+
+        event_list:array                    list of event names and properties
+        options:object                      optional batch configuration
+        callback:function(error_list:array) callback is called when the request is
+                                            finished or an error occurs
+    */
+    metrics.import_batch = function(event_list, options, callback) {
+        var MP_BATCH_SIZE = 50; // Mixpanel API permits 50 events per request
+
+        var data_length = event_list.length,
+            total_requests = Math.ceil(data_length / MP_BATCH_SIZE),
+            max_simultaneous_requests = total_requests,
+            completed_requests = 0,
+            event_group_idx = 0,
+            request_errors = [];
+
+        if (typeof(options) === 'function' || !options) {
+            callback = options;
+            options = {};
+        }
+        if (options.max_batch_size) {
+            max_simultaneous_requests = Math.min(
+                total_requests, Math.max(Math.floor(options.max_batch_size / MP_BATCH_SIZE), 1)
```

Comment:
Yeah, I was working with the assumption that a batch size of <50 could be safely ignored; but I suppose if someone has a reason for wanting smaller batches we should honor it.

I've also refactored the calculations in this method to get rid of all of the request-counting and instead go solely on events sent and to be sent, since we're calculating that anyway; no need for two sources of truth.